pyccel/stdlib/gFTL_functions/Vector_extensions.inc

```diff
@@ -77,4 +77,29 @@ contains
 
   end function __IDENTITY(Vector)_max
 
+  subroutine __IDENTITY(Vector)_insert(my_vector, idx, value)
+    use, intrinsic :: ISO_C_BINDING, only: i64 => C_INT64_T
+
+    class(Vector), intent(inout)              :: my_vector
+    integer(i64), intent(in)                  :: idx
+    __T_declare_dummy__, intent(in)           :: value
+  
+    integer(i64) :: n, pos, i
+  
+    n = my_vector%size()
+    pos = idx
+  
+    if (pos < 0) pos = pos + n
```

Comment:
Why write a new method instead of using https://github.com/Goddard-Fortran-Ecosystem/gFTL/blob/main/include/v2/vector/vector.md#insert ?

Reply:
I assumed negative indices would work in the same way as for arrays

Reply:
insert is very lenient with indices. the index can be way larger than the size of the array, then it would be appended, or too small (negative) then it would be prepended to the array. maybe keep this subroutine as a wrapper around their insert?

Reply:
Ah you're right! I think we even have tests for some of these awkward insides. In that case a function that fixes the indices then calls the existing function sounds perfect